charts/argo-cd/templates/argocd-server/alb-grpc-service.yaml

```diff
@@ -20,5 +20,5 @@ spec:
   selector:
     {{- include "argo-cd.selectorLabels" (dict "context" . "name" .Values.server.name) | nindent 4 }}
   sessionAffinity: None
-  type: ClusterIP
+  type: NodePort
```

Comment:
How come we'd need NodePort here? Upstream is not mentioning this: https://argoproj.github.io/argo-cd/operator-manual/ingress/#aws-application-load-balancers-albs-and-classic-elb-http-mode :thinking:

Reply:
AWS Load Balancer controller in the latest version return an error saying that can't map a targetGroup to a service without a port. This occur because ClusterIP type don't open a port in hosts to be used as destination to targetGroup. Because of this, AWS Controller stop ALB creation.

```
{"level":"error","ts":1628462993.0268402,"logger":"controller","msg":"Reconciler error","reconcilerGroup":"elbv2.k8s.aws","reconcilerKind":"TargetGroupBinding","controller":"targetGroupBinding","name":"k8s-argocd-argocdse-380b120770","namespace":"argocd"
,"error":"service type must be either 'NodePort' or 'LoadBalancer': argocd/argocd-server-grpc"}
```

Reply:
Maybe we need to update that docs. I will discuss about this in the argo-cd repo.

Reply:
Seeing here, the docs in master branch already updated, but not available in latest page.

https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/ingress.md#aws-application-load-balancers-albs-and-classic-elb-http-mode

Reply:
Well spotted :+1: